### DISCUSSION TITLE
New Controller Design

### DISCUSSION THREAD

**Discussion by derwin12** [xLightsSequencer/xLights#4541] [Ideas]
Thoughts on moving the buttons to icon/bitmap buttons along the top. Allowing us to better utilize the space along the sides.
Which size could we key off - the size of the top icons or the effect icons?
The right side could be locked at a min/max size based on the size of the buttons or whatever works there.

![image](https://github.com/xLightsSequencer/xLights/assets/4643499/aece0920-0205-4fc9-b243-cafd49a3b8e5)

Further .. perhaps add an icon for the controllers -- just for fun :-) A Kulp Logo/Falcon Logo/Scott Nation logo?? :-)

**Comment by derwin12** [comment]
Another is the groups ..
Remove some of the redundancy on the filter section.. reduce the width to allow more space for the actual models to be included in the group.
![image](https://github.com/xLightsSequencer/xLights/assets/4643499/db78a6ae-e070-4e07-8ff4-3ee287686bcd)

**Comment by cybercop23** [comment]
I think these are relatively small and easy changes to enhance the user experience. We see lots of folks (especially new to the hobby) that have a very hard time navigating and sections of those screen unecessaraly disporpotional. Condensing them and even locking some in place (ala Auda
city) willl be a great help.

**Comment by marklancer** [comment]
If you need SVG's for the buttons, let me know and I can make them.

**Reply by cybercop23** [reply]
If you would want to make some small ones that look like the effect ones, we're happy to take them... :)

**Reply by marklancer** [reply]
Attached are some SVG's to start testing with.  Color changes would be quick so if these work we can work on a colors.  This set was based on Dark Mode version.
![Delete](https://github.com/xLightsSequencer/xLights/assets/59756293/33f06ebe-abb9-4393-a100-a3f607c5e2ff)
![Open](https://github.com/xLightsSequencer/xLights/assets/59756293/901b545e-078c-4d9e-8ace-14e6ece16e78)
![Open_Proxy](https://github.com/xLightsSequencer/xLights/assets/59756293/6fa0b369-702f-40b7-b60a-8e4b1bfc86ff)
![Upload_Inputs](https://github.com/xLightsSequencer/xLights/assets/59756293/9e9c9ad8-b4f9-436f-8597-29c233e375a5)
![Upload_Outputs](https://github.com/xLightsSequencer/xLights/assets/59756293/2eae2e6a-68ed-4c87-a197-56d299607656)
![Visualize](https://github.com/xLightsSequencer/xLights/assets/59756293/86a9db47-18c3-48f6-81a3-06a49b0be1c5)

**Reply by AzGilrock** [reply]
I'm not a good one to access this because I'm not a fan of the flat look most people do with SVG's.  I've done a lot of vector graphics and my stuff uses linear and radial gradients so things look interesting and not flat.

**Reply by AzGilrock** [reply]
This is artwork I worked on for my arcade cabinet.  All done with vector graphics.  Some of the stuff that looks like textures is the filter capability in InkScape so that's all driven by mathmatical algorithms.  But all the other stuff you can see where I like to add gradients to make things look like they have depth.

![dlair](https://github.com/xLightsSequencer/xLights/assets/2024669/34f8fd6a-13e8-4798-9911-e3031c3ac4b4)

**Comment by marklancer** [comment]
I'm totally open to making them look better, I just threw something together quick to see if the file format works.

We need a matrix plugin for Dragon's Lair.

**Reply by AzGilrock** [reply]
Trust me I've thought about adding piMame support to FPP and have kids be able to play on a matrix with a big button control panel.
